Proposals/0002-date-interval-differences.md

```diff
@@ -0,0 +1,66 @@
+# Date Interval Differences
+
+* Proposal: [FOU-0002](0002-date-interval-differences.md)
+* Authors: [Kiel Gillard](https://github.com/kielgillard)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [apple/swift-foundation#330](https://github.com/apple/swift-foundation/pull/330)
+* Review: ([pitch](https://forums.swift.org/...))
+
+## Introduction
+
+Add small but convenient, quality-assured methods on `DateInterval` for computing time and date intervals between other date intervals.
+
+## Motivation
+
+`DateInterval` could offer more APIs to help developers work with ranges of absolute moments in time. For example, a developer might be building a log book, shift planner, time tracker or implementing a time based feature in a less specialised app. In the more complex cases, an app may require users to enter start and end times for particular activities, arrange them in a particular sequence with some acceptable gap between (e.g. work here for 2 hours, work there for 1.5 hours and rest for 10 minutes with at most five minutes between, no overlaps, etc). This app would need to validate the intervals formed by the dates entered by at least: checking the intersections of intervals, measuring how much those intervals intersect by and/or by measuring the time between them. In the case of a simpler feature, displaying the time until or since an event, with different formatting or app behaviour on either side (e.g. a library book due/overdue).
+
+`DateInterval` would be an intuitive type to begin such implementations with. However, developers would quickly hit a limit with its out-of-the-box utility and become responsible for providing what is missing. Further, the precedent set by other Foundation types have shaped our intuitions about what utility we could expect of it.
+
+## Proposed solution
+
+I would like to propose a set of small but delightful, quality-assured additions to `DateInterval` for computing the time or date interval between two, potentially non-intersecting date intervals: `timeIntervalSince(_:)`, `dateIntervalSince(_:)`. I would propose two overloads for these APIs: one taking `DateInterval` values and another taking `Date` values.
```

Comment:
Same comment here -- it's useful to show an example for the proposed API in code